scipy/signal/tests/test_signaltools.py

```diff
@@ -1115,7 +1106,7 @@ def test_invalid_dtypes(self, dtype):
     def test_none(self):
         # gh-1651, trac #1124. Ensure this does not segfault.
         with pytest.warns(UserWarning):
-            assert_raises(TypeError, signal.medfilt, None)
+            assert_raises(ValueError, signal.medfilt, None)
```

Comment:
Looks like there was some sort of special casing here for None. Have it lumped in with the others that raise ValueError but could special case so this remains a TypeError